packages/client/README.md

```diff
@@ -550,137 +664,100 @@ const dataUnion = await client.deployDataUnion({
+The static function `StreamrClient.generateEthereumAccount()` generates a new Ethereum private key and returns an object with fields `address` and `privateKey`. Note that this private key can be used to authenticate to the Streamr API by passing it in the authentication options, as described earlier in this document.
+```js 
+const wallet = StreamrClient.generateEthereumAccount()
+```
+Generates a random Ethereum account object:
+```js
+    wallet = {address, privateKey}
+```
+In order to retrieve the client's address an async call must me made to `streamr.getAddress`
+```js
+const address = await streamr.getAddress()
+```
 
-| Name                                    | Returns                 |   Description    |
-| :-------------------------------------- | :---------------------- | :--------------- |
-| `*` generateEthereumAccount()           | `{address, privatekey}` | Generates a random Ethereum account  |
-| getTokenBalance(address)                | `BigNumber`             | Mainnet DATA token balance |
-| getSidechainTokenBalance(address)       | `BigNumber`             | Sidechain DATA token balance |
+## Advanced Usage
 
-`*` The static function `StreamrClient.generateEthereumAccount()` generates a new
-Ethereum private key and returns an object with fields `address` and `privateKey`.
-Note that this private key can be used to authenticate to the Streamr API
-by passing it in the authentication options, as described earlier in this document.
 
-## Events
+### Manual connection management
 
-The client and the subscriptions can fire events as detailed below.
-You can bind to them using `on`.
+By default the client will automatically connect and disconnect as needed, ideally you should not need to manage connection state explicitly.
 
-| Name                                | Description                                                                              |
-| :---------------------------------- | :--------------------------------------------------------------------------------------- |
-| on(eventName, function)             | Binds a `function` to an event called `eventName`                                        |
-| once(eventName, function)           | Binds a `function` to an event called `eventName`. It gets called once and then removed. |
-| removeListener(eventName, function) | Unbinds the `function` from events called `eventName`                                    |
 
-### Events on the StreamrClient instance
+Specifically, it will automatically connect when you publish or subscribe, and automatically disconnect once all subscriptions are removed and no messages were recently published. This behaviour can be disabled using the `autoConnect` & `autoDisconnect` options when creating a `new StreamrClient`. Explicit calls to either `connect()` or `disconnect()` will disable all `autoConnect` & `autoDisconnect` functionality, but they can be re-enabled by calling `enableAutoConnect()` or `enableAutoDisconnect()`.
```

Comment:
Is this description still valid? To me it looks like `disconnect` is an alias to `destroy`, and therefore I'd expect we shouldn't do anything after destroying the instance:
https://github.com/streamr-dev/network-monorepo/blob/557a7ecc92779e1a40d6d6d5e7485bc4a40516d9/packages/client/src/StreamrClient.ts#L147

Also, as the `disconnect` is a deprecated method, we should use `destroy` instead

Reply:
should the references to `disconnect` be removed and the text reworked into "here's how to auto-connect" or should the section be removed?

Reply:
No. Two things:
- `Disconnect` is a deprecated method, and the documentation should not reference to any deprecated methods. A replacement for the method is `destroy()`. (see the link in the previous message)
- To me it sounds that destroy() is not a direct replacement for disconnect() because the word "disconnect" would tell me that I can reconnect if I want. But a word "destroy" indicated that after destroying a client, we should't interact with that client instance anymore.

If the assumption in my second bullet-point is valid, some sentences in the above section needs to be corrected. Maybe easiest to test this out in practice?

Reply:
Removed the references to `disconnect` in the text and snippets, not sure this is the right place to mention `destroy` since it is unrelated to the connect/disconnect functionality.

Here's the latest text to look into, am assuming it will require a few more corrections:
https://github.com/streamr-dev/network-monorepo/blob/1119d4159e55bcbacc59d119cac3c23aeb08b443/packages/client/README.md#manual-connection-management

Reply:
o me it looks like `autoConnect` and `autoDisconnect` don't do anything. Have you tried to execute the example? Or is there an example in the examples repo about this?

Reply:
Made a PR: https://github.com/streamr-dev/network-monorepo/pull/471